src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -1328,8 +1328,8 @@ char* MetaspaceShared::reserve_address_space_for_archives(FileMapInfo* static_ma
       total_space_rs = ReservedSpace(total_range_size, archive_space_alignment,
                                      os::vm_page_size(), (char*) base_address);
     } else {
-      // Reserve at any address, but leave it up to the platform to choose a good one.
-      total_space_rs = Metaspace::reserve_address_space_for_compressed_classes(total_range_size);
+      // Reserve anywhere, but mapping start address must be directly encodable as encoding base.
+      total_space_rs = Metaspace::reserve_address_space_for_compressed_classes(total_range_size, true);
```

Comment:
That's why I favored my original name for that parameter, "strict_base", since it much clearer described behavior than "cds_runtime" does.

As for split-up, I rather avoid it. The resulting parts would be difficult to describe and overlap in functionality. The cds_runtime path, including this parameter, will hopefully be removed soon when we implement recalculating the nKlassIDs at CDS runtime.